rust/processor/src/processors/marketplaces_processor.rs

```diff
@@ -0,0 +1,1048 @@
+            // and collection address for token v2.
+            let mut collection_metadatas: HashMap<String, CollectionMetadata> =
+                HashMap::new();
+
+            let mut collection_offer_filled_metadatas: HashMap<
+                String,
+                CollectionOfferEventMetadata,
+            > = HashMap::new();
+
+            // Loop through the events for the all activities
+            for (index, event) in inner.events.iter().enumerate() {
+                // get event type and contract address
+                let type_str_vec = event.type_str.as_str().split("::").collect::<Vec<&str>>();
+                let contract_addr = type_str_vec[0];
+                let event_type = type_str_vec[type_str_vec.len() - 1];
+
+                // Only the events in the set will be indexed
+                // TODO - We currently only index events from our own smart contracts
+                if !events_set.contains(event_type) || !contract_addr.eq(marketplace_contract_address) {
+                    continue;
+                }
+
+                let event_data = serde_json::from_str::<serde_json::Value>(&event.data).unwrap();
+                let activity_fee_schedule_id = event.key.as_ref().unwrap().account_address.as_str();
+                let price = event_data["price"]
+                    .as_str()
+                    .expect("Error: Price missing from marketplace activity")
+                    .parse::<BigDecimal>()
+                    .unwrap();
+                let activity_offer_or_listing_id =
+                    if let Some(listing) = event_data["listing"].as_str() {
+                        standardize_address(listing)
+                    } else if let Some(token_offer) = event_data["token_offer"].as_str() {
+                        standardize_address(token_offer)
+                    } else {
+                        // collection_offer
+                        standardize_address(event_data["collection_offer"].as_str().unwrap())
+                    };
+                
+                // Get the token and collection metadata from event.
+                let token_metadata = TokenMetadata::from_event(event, txn_version).unwrap();
+                if let Some(token_metadata) = token_metadata {
+                    token_metadatas.insert(token_metadata.get_token_address().unwrap(), token_metadata.clone());
+                }
+                let collection_metadata =
+                    CollectionMetadata::from_event(event, txn_version).unwrap();
+                if let Some(collection_metadata) = collection_metadata {
+                    collection_metadatas.insert(
+                        collection_metadata.get_collection_address().unwrap(),
+                        collection_metadata,
+                    );
+                }
+
+                // Get the acitivty, listing, token_offer, and auction from event.
```

Comment:
Do something like [this](https://github.com/aptos-labs/aptos-indexer-processors/blob/b2325e585d68c5e2089527504943445906b4711d/rust/processor/src/models/token_v2_models/v2_token_utils.rs#L370), so that if it matches a condition, it won't run the next statements.